src/platform/backends/lxd/lxd_vm_image_vault.cpp

```diff
@@ -81,42 +81,37 @@ mp::VMImage mp::LXDVMImageVault::fetch_image(const FetchType& fetch_type, const
     {
         auto instance_info = lxd_request(
             manager, "GET",
-            QUrl(QString("%1/virtual_machines/%2").arg(base_url.toString()).arg(QString::fromStdString(query.name))));
+            QUrl(QString("%1/virtual-machines/%2").arg(base_url.toString()).arg(QString::fromStdString(query.name))));
 
         auto id = instance_info["metadata"].toObject()["config"].toObject()["volatile.base_image"].toString();
 
         for (const auto& image_host : image_hosts)
         {
-            try
-            {
-                auto info = image_host->info_for_full_hash(id.toStdString());
+            auto info = image_host->info_for_full_hash(id.toStdString());
 
-                VMImage source_image;
+            VMImage source_image;
 
-                source_image.id = id.toStdString();
-                source_image.stream_location = info.stream_location.toStdString();
-                source_image.original_release = info.release_title.toStdString();
-                source_image.release_date = info.version.toStdString();
+            source_image.id = id.toStdString();
+            source_image.stream_location = info.stream_location.toStdString();
+            source_image.original_release = info.release_title.toStdString();
+            source_image.release_date = info.version.toStdString();
 
-                for (const auto& alias : info.aliases)
-                {
-                    source_image.aliases.push_back(alias.toStdString());
-                }
-            }
-            catch (const std::exception&)
+            for (const auto& alias : info.aliases)
             {
-                // Nothing to do, just move on
+                source_image.aliases.push_back(alias.toStdString());
             }
+
+            return source_image;
         }
     }
     catch (const std::exception&)
     {
-        // Instance doesn't exist so move on
+        // Instance or image doesn't exist so move on
     }
```

Comment:
Hmm, I could.  'lxd_request()` will throw `LXDNotFoundException` which I could catch (and ignore) separately.  However, I would have to modify `UbuntuVMImageHost::info_for_full_hash_impl()` to throw a new type if an image is not found.  Since that affects all other backends, is this the place to add that?

Reply:
Actually, if I do add a different exception type when an image isn't found, I would have to modify `DefaultVMImageVault` instead of "all other backends".  Still, is this PR the place to add this since I would be touching an unrelated file?  I would rather add the different exception type in a follow up PR.  That said, I can add another `catch` here for `LXDNotFoundException` in anticipation of this.

Reply:
@Saviq 

I added the `LXDNotFoundException` `catch` block.  Let me know what you think about my proposal for the new exception type.